_posts/2017-02-15-using-stuntman-with-aspnet-mvc-video.md

```diff
@@ -0,0 +1,143 @@
+
+The first is accessing the user in an MVC Action. 
+
+```csharp
+/* MVC */
+public ActionResult Index() {
+    /* used here */
+    return Content(User.Identity.Name);
+}
+```
+
+The second is accessing the user via MVC authorized attribute.
+
+```csharp
+/* yep, here too */
+[Authorize(Roles="Admin")] 
+public ActionResult Secure() {
+    return Content("Secure");
+}
+```
+
+And finally the third example is accessing the user's name in an MVC view.
+
+```html
+<div class="username">
+    <!-- your views too! -->
+    @User.Identity.Name
+</div>
+```
+
+All three scenarios are leveraging IPrincipal and IIdentity.
+
+
+Now that we've answered our first question of "how does ASP.NET authentication work?" we begin to explore how Stuntman works to accelerate your local development process. 
+
+So how does Stuntman work? Let's start with your basic ASP.NET MVC example. I'll first start by going into the new project dialog and creating a web application named "BasicExample". Then i'll select MVC as my starter template. Once my application is initialized, I will need to add references via Nuget.
+
+The first reference is OWIN, since we're hosting in ASP.NET MVC application will
+need `Microsoft.Owin.Host.SystemWeb`.
+
+```console
+> Install-Package Microsoft.Owin.Host.SystemWeb
+```
+ 
+The second reference will need is Stuntman. So we'll search for Stuntman on Nuget.
+
+```console
+> Install-Package RimDev.Stuntman
+```
+
+Once we find Stuntman, we'll just go ahead and install it just like we did with OWIN. Now that we have both Stuntman and OWIN
+installed we'll need a `Startup.cs`. The Startup.cs class tells OWIN how to configure our asp.net OWIN enabled
+MVC application
+
+I've already prepared a `Startup.cs` class which have copied from the GitHub readme of Stuntman. A few things to note about our startup.cs. First, `StuntmanOptions` are created statically at the top of our startup class. Secondly, inside of our configuration method we set up a user in our `StuntmanOptions` users "user 1" with a set of
+given claims. Finally, we say `app.UseStuntman()` passing in the Stuntman options to make Stuntman visible to our
+UI. 
+
+```csharp
+// OWIN Startup class
+public class Startup
+{
+    public static readonly StuntmanOptions StuntmanOptions = new StuntmanOptions();
+
+    public void Configuration(IAppBuilder app)
+    {
+        StuntmanOptions
+            .AddUser(new StuntmanUser("user-1", "User 1")
+                .AddClaim("given_name", "John")
+                .AddClaim("family_name", "Doe"));
+    
+            app.UseStuntman(StuntmanOptions);        
+    }
+}
+
+```
+
+We need to alter our `_Layout` file. I first want to start by modifying our home link and exposing our
```

Comment:
You may want to add `.cshtml` to `_Layout` here? Although, if it's a transcript, not sure how that works? :smiley: